src/fenwick.jl

```diff
@@ -0,0 +1,96 @@
+import Base.getindex
+"""
+    FenwickTree{T}(n)
+    
+Constructs a [`FenwickTree`](https://en.wikipedia.org/wiki/Fenwick_tree) of length `n`.
+ 
+"""
+mutable struct FenwickTree{T}
+    bit::Vector{T}
+    n::Integer
+end
+
+FenwickTree{T}() where T = FenwickTree{T}(0)
+FenwickTree{T}(n::Integer) where T = FenwickTree{T}(zeros(T, n), n)
+
+"""
+    FenwickTree(arr::AbstractArray) 
+    
+Constructs a [`FenwickTree`](https://en.wikipedia.org/wiki/Fenwick_tree) from `arr`.
+ 
+"""
+function FenwickTree(a::T) where T <: AbstractVector
+    U = eltype(a)
+    n = size(a)[1]
+    bit = FenwickTree{U}(n)
+    @boundscheck for i = 1:n
+        update!(bit, i, a[i])
+    end
+    bit
+end
+
+bit(F::FenwickTree{T}) where T = F.bit
```

Comment:
```suggestion
bit(F::FenwickTree) = F.bit
```
Type parameter isn't used